### DISCUSSION TITLE
OpenGlControlBase uses left handed coordinate system by default?

### DISCUSSION THREAD

**Discussion by jameson75** [AvaloniaUI/Avalonia#6952] [Q&A]
Hey everyone. I started using OpenGlControlBase to implement a custom control and noticed something peculiar. It seems like the coordinate system is left handed by default, with the y axis pointing down. I expected the coordinate system to be right handed with the y-axis pointing up. Is this by design?... or maybe I'm just missing something?

I did read in another thread that AvaloniaUI uses ANGLE (an OpenGL ES implementation on top of DirectX) for its OpenGL client - could this be the reason for the left-handed coordinate system?

**Comment by kekekeks** [comment]
It might actually be just a flipped texture during the composition step.

**Reply by jameson75** [reply]
Thanks for the feedback. Appreciated.

A little background... 
My starting point was the sample provided here https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/OpenGlPage.xaml.cs.
Notice on [line 349](https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/OpenGlPage.xaml.cs#L349) the camera's up vector is (0,-1,0) so that the teapot model is rendered upright. 

I'm not sure where the composition step takes place.

I guess a texture flipped along the x-axis might explain why _y_ is pointing down. It still leaves me curious about the "handedness" of the coordinate system. Any ideas why  the z-axis would be pointing away from the near plane?

Thanks.

**Reply by kekekeks** [reply]
We are importing that OpenGL surface to Skia later. There is a parameter that controls an Y flip. It's possible that I just somehow misconfigured the OpenGL context and then "fixed" that by flipping the texture. Or just flipped the texture when I shouldn't. 

Could you check if the same drawing code produces different results when using OpenTK?
